sklearn/linear_model/_ridge.py

```diff
@@ -157,12 +157,13 @@ def _solve_cholesky(X, y, alpha):
 
     if one_alpha:
         A.flat[:: n_features + 1] += alpha[0]
-        return linalg.solve(A, Xy, sym_pos=True, overwrite_a=True).T
+        return linalg.solve(A, Xy, sym_pos=True, overwrite_a=True, check_finite=False).T
```

Comment:
`Xy` is computed using `safe_sparse_dot` and even if `X.Y` and `y` are both finite, `Xy` may not be finite. See https://github.com/scikit-learn/scikit-learn/issues/18837#issuecomment-1012239084